examples/kg-chat/csv_kg_chat.py

```diff
@@ -0,0 +1,266 @@
+    DataFrame: A DataFrame containing the data loaded from the CSV file.
+
+    Raises:
+    Exception: An error occurred while trying to read the CSV file.
+
+    Note:
+    The function assumes the CSV file has a header row.
+    """
+    try:
+        df = read_csv(csv_location)
+        return df
+    except Exception as e:
+        raise Exception(f"Error occurred while reading the CSV file: {e}")
+
+
+def _preprocess_dataframe_for_neo4j(
+    df: DataFrame, default_value: str = None, remove_null_rows: bool = True
+) -> DataFrame:
+    """
+    Preprocess a DataFrame for Neo4j import by fixing mismatched quotes in string
+        columns and handling null or missing values.
+
+    Args:
+        df (DataFrame): The DataFrame to be preprocessed.
+        default_value (str, optional): The default value to replace null values.
+        This is ignored if remove_null_rows is True. Defaults to None.
+        remove_null_rows (bool, optional): If True, rows with any null values will
+            be removed.
+        If False, null values will be filled with default_value. Defaults to False.
+
+    Returns:
+        DataFrame: The preprocessed DataFrame ready for Neo4j import.
+    """
+
+    # Fix mismatched quotes in string columns
+    for column in df.select_dtypes(include=["object"]):
+        df[column] = df[column].apply(
+            lambda x: x + '"' if (isinstance(x, str) and x.count('"') % 2 != 0) else x
+        )
+
+    # Handle null or missing values
+    if remove_null_rows:
+        df = df.dropna()
+    else:
+        if default_value is not None:
+            df = df.fillna(default_value)
+
+    return df
+
+
+class PandasToKGTool(ToolMessage):
+    request: str = "pandas_to_kg"
+    purpose: str = """Use this tool to create ONLY nodes and their relashipnships based
+    on the created model.
+    Take into account that the Cypher query will be executed while iterating the rows in
+    the CSV file (like this `index, row in df.iterrows()`). So NO NEED to load the CSV.
+    Make sure you send me the cypher query in this format: 
+    - placehoders in <cypherQuery> should be based on the CSV header. 
+    - <args> an array wherein each element corresponds to every placeholder in the 
+    <cypherQuery> and provided in the same order as the headers. 
+    SO the <args> should be like this: `[row_dict[header] for header in headers]`
+    """
+    cypherQuery: str
+    args: list[str]
+
+    @classmethod
+    def examples(cls) -> List["ToolMessage"]:
+        return [
+            cls(
+                cypherQuery="""MERGE (employee:Employee {name: $employeeName, id: $employeeId})\n
+                MERGE (department:Department {name: $departmentName})\n
+                MERGE (employee)-[:WORKS_IN]->(department)\n
+                SET employee.email = $employeeEmail""",
+                args=["employeeName", "employeeId", "departmentName", "employeeEmail"],
+            ),
+        ]
+
+
+class CSVChatGraphAgent(Neo4jChatAgent):
```

Comment:
worth adding this Agent and `PandasToKGTool` to the main branch, since these can be generally useful?

Reply:
yes, let's do that if there is no further modifications

Reply:
Ok please move these

Reply:
I didn't get it, moving them to where?

Reply:
I meant move these out of examples and into the core codebase, adjacent to `Neo4jChatAgent`